Unhandled Promise Rejections in compileTypeScript

## Current Behavior
Typescript libraries with errors cause build to hang and not complete due to unhandled promise rejections.

## Expected Behavior
Successful builds.

## Steps to Reproduce
```bash
npx create-nx-workspace test-ts
cd test-ts
npm i -D @nrwl/js
nx generate @nrwl/js:library --name=test --buildable
echo "let error =;" >> libs/test/src/index.ts
nx build test
```
(or `nx affected:build`, but no logs printed)

### Failure Logs
```
(node:3718) UnhandledPromiseRejectionWarning: Error: libs/test/src/index.ts:4:12 - error TS1109: Expression expected.

4 let error =;
             ~

    at createProgram (/home/jack/dev/Node/test-ts/node_modules/@nrwl/workspace/src/utilities/typescript/compilation.js:99:15)
    at compileTypeScript (/home/jack/dev/Node/test-ts/node_modules/@nrwl/workspace/src/utilities/typescript/compilation.js:15:12)
    at /home/jack/dev/Node/test-ts/node_modules/@nrwl/js/src/utils/typescript/compile-typescript-files.js:37:73
    at Generator.next (<anonymous>)
    at /home/jack/dev/Node/test-ts/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
    at __awaiter (/home/jack/dev/Node/test-ts/node_modules/tslib/tslib.js:113:16)
    at /home/jack/dev/Node/test-ts/node_modules/@nrwl/js/src/utils/typescript/compile-typescript-files.js:28:227
    at Object.[Symbol.asyncIterator] (/home/jack/dev/Node/test-ts/node_modules/@nrwl/js/src/utils/create-async-iterable/create-async-iteratable.js:11:13)
    at __asyncValues (/home/jack/dev/Node/test-ts/node_modules/tslib/tslib.js:241:22)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:3718) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:3718) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

### Environment

(NX installed globally, version 13.8.3)
(also fails using Node 12)

```
 >  NX   Report complete - copy this into the issue template

   Node : 14.19.0
   OS   : linux x64
   npm  : 6.14.16

   nx : undefined
   @nrwl/angular : undefined
   @nrwl/cli : 13.8.3
   @nrwl/cypress : undefined
   @nrwl/detox : undefined
   @nrwl/devkit : 13.8.3
   @nrwl/eslint-plugin-nx : 13.8.3
   @nrwl/express : undefined
   @nrwl/jest : 13.8.3
   @nrwl/js : 13.8.3
   @nrwl/linter : 13.8.3
   @nrwl/nest : undefined
   @nrwl/next : undefined
   @nrwl/node : undefined
   @nrwl/nx-cloud : undefined
   @nrwl/react : undefined
   @nrwl/react-native : undefined
   @nrwl/schematics : undefined
   @nrwl/storybook : undefined
   @nrwl/tao : 13.8.3
   @nrwl/web : undefined
   @nrwl/workspace : 13.8.3
   typescript : 4.5.5
   rxjs : 6.6.7
   ---------------------------------------
   Community plugins:
```

--- a/packages/workspace/src/utilities/typescript/compilation.ts
+++ b/packages/workspace/src/utilities/typescript/compilation.ts
@@ -171,7 +171,7 @@ function createProgram(
       }
     );
     logger.error(diagnostics);
-    throw new Error(diagnostics);
+    return { success: false };
   } else {
     logger.info(
       `Done compiling TypeScript files for project "${projectName}".`

